kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -56,6 +56,39 @@ class RobotState;
 namespace kinematics
 {
 
+/*
+ * @enum KinematicSearches
+ * @brief Flags for choosing the type discretization applied on the redundancy during a multiple ik solution search
+ */
+namespace KinematicSearches
+{
+  enum KinematicSearch
+  {
+    ONE = 1,
+    ALL_DISCRETIZED ,
+    SOME_DISCRETIZED,
+    RANDOM_SAMPLE,
+    ALL
+  };
+}
+typedef KinematicSearches::KinematicSearch KinematicSearch;
+
+/*
+ * @enum KinematicErrors
+ * @brief Kinematic error codes that occur in a multiple ik solution search
+ */
+namespace KinematicErrors
+{
+  enum KinematicError
+  {
+    OK = 1,
+    UNSUPORTED_SEARCH_REQUESTED,
+    DISCRETIZATION_NOT_INITIALIZED,
+    NO_SOLUTION
```

Comment:
@sachinchitta, can you provide a link to the error flags (google can't seem to find them for me).  Also, can you suggest other error conditions that we should consider?